portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,7 +35,46 @@
 public class DataServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Joachim!</h1>");
+    Gson gson = new Gson();
+
+    ArrayList<String> data = new ArrayList<String>(0);
+    Query query = new Query("Comments");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
As things get more complicated, it'll become more and more helpful to keep your code organized in small functions that focus on doing one thing.  It might be good to extract this part out as a separate method, maybe `fetchCommentsAsList(query)`.

You could also consider extracting out the query creation and the setting of the JSON response as well.  That would allow you to keep a similar level of abstraction throughout this function, which will aid in readability.

Take a look at the following:
```
  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
    sendJsonResponse(response, convertCommentListToJson(
        fetchCommentsAsList(createQueryToFetchAllComments())));
  }

  private void sendJsonResponse(HttpServletResponse response, String json) {
    response.setContentType("application/json;");
    response.getWriter().println(json);
  }

  private String convertCommentListToJson(List<String> list) {
    ....
  }

  private List<String> fetchCommentsAsList(Query query) {
    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
    List<String> commentsList = new ArrayList<String>();
    for (Entity entity : datastore.prepare(query).asIterable()) {
      String comment = (String) entity.getProperty("comment");
      commentsList.add(comment);
    }
    return commentsList;
  }

  private Query createQueryToFetchAllComments() {
    return new Query(COMMENTS_ENTITY);
  }
```

The entry point, the doGet method, becomes very easy to understand at a glance since it's a series of higher-level method calls that clearly describe what is happening:

- create a query that will fetch all the comments
- get a list of all of the comments for the query
- convert that list of comments to json
- send that json response to the client

All of the details about these operations are hidden in their respective methods so those details don't obscure what's happening.  Additionally, this allows you to more easily reuse parts, such as the sendJsonResponse method, which could be extracted to a helper library and used in multiple servlets.